Celery options

omerfeinberg opened the pull request:
This branch allows usage of options available for celery tasks: priority, serializer, compression, etc.
 
Great job with this package! it performs incredibly well

Commit:
adds possibility to include user defined dict in serialized nodes

Commit:
adds same thing for Workflow

Commit:
update README and add example

omerfeinberg committed:
celery_options

omerfeinberg committed:
celery_options

omerfeinberg committed:
main commit

omerfeinberg committed:
main commit

omerfeinberg committed:
line lenght fixes

omerfeinberg committed:
line lenght fixes

omerfeinberg committed:
Bump version: 0.7.0 → 0.8.0

omerfeinberg committed:
Bump version: 0.7.0 → 0.8.0

brabiega commented:
Hello! Thanks a lot for this contribution. I'll try to take a look tomorrow or the day after. Looks quite good!

brabiega commented on celery_dyrygent/workflows/workflow.py:
```diff
@@ -605,15 +619,20 @@ def freeze(self):
         return signature
 
-    def apply_async(self):
+    def apply_async(self, **kwargs):
```
Perhaps it would be better to use
```
def apply_async(self, options=None)
```
So in the future it would be possible to pass some additional arguments.
What do you think?

omerfeinberg replied:
Thanks for the quick feedback man! 
So regarding both your comments, that's pretty much the only thing I wasn't sure about. I think we should support apply_async with the same signature as celery, which means using **kwargs. I don't think it should hurt us in the future while making changes, but I'm not 100% about that, so whatever you think :)

(Note that while initializing a Workflow object I do use options=dict(...), so your suggestion has the advantage of uniformity)

brabiega replied:
I know celery uses **kwargs, but I don't think that's a good design decision.
Let's go with options=dict() :)

omerfeinberg replied:
no problem, changed :)

omerfeinberg replied:
Hey man, waiting for your approval :)

brabiega replied:
Sure thing. 
I'd like to merge https://github.com/ovh/celery-dyrygent/pull/6 first.
And then release both changes under 0.8.0.
Besides I'm testing your changes right now :)

Edit:
changes required due to conflict are minimal

omerfeinberg replied:
Cool, they were indeed minimal.
I changed custom_payload to use the straight_serialzible default mechanism, I think it's neater.
No other changes

brabiega replied:
Thx. The straight_serialzible default mechanism is pretty cool.

brabiega commented on README.md:
````diff
@@ -123,6 +123,17 @@ Handler params are:
+``` python
+wf = Workflow(options={'priority': 10})
+wf.add_celery_canvas(canvas)
+wf.apply_async(priority=8)
````
I'm wondering if it would be better to specify options directly.
```
wf.apply_async(options=dict(priority=8))
```

brabiega reviewed (commented):
Just a question about passing options to wf.apply_async. See comments

omerfeinberg committed:
made apply_async's options explicit

omerfeinberg committed:
made apply_async's options explicit

Commit:
Code Review :

- Rename added dict from user_params to custom_payload
- simplify ser/deser by removing always including custom_payload
- replace ifs by getter with default

Commit:
Tests adjustments

brabiega committed:
Merge pull request #6 from abak/user_params

adds possibility to include user defined dict in serialized nodes

omerfeinberg committed:
integrated custom payload

omerfeinberg committed:
straight serializable custom payload

brabiega reviewed (approved):
Looks good. I'll just do some final tests tomorrow morning.
